6.5.set.py

```diff
@@ -0,0 +1,43 @@
+
+import random
+from itertools import combinations
+
+def create_cards():
+    cards = []
+
+    for shape in range(1, 4):
+        for shade in range(1, 4):
+            for color in range(1, 4):
+                for count in range(1, 4):
+                    cards.append({'shape': shape, 'shade': shade, 'color': color, 'count': count})
+
+    return cards
+
+
+def is_set(trio):
+    count_sum = sum([card['count'] for card in trio])
+    shape_sum = sum([card['shape'] for card in trio])
+    shade_sum = sum([card['shade'] for card in trio])
+    color_sum = sum([card['color'] for card in trio])
+    set_values_mod3 = [count_sum % 3, shape_sum % 3, shade_sum % 3, color_sum % 3]
+    return set_values_mod3 == [0, 0, 0, 0]
+
+def sets():
+    cards = create_cards()
+    random.shuffle(cards)
+
+    twelve = cards[:12]
+    sets = []
+    for trio in combinations(twelve, 3):
+        if is_set(trio):
+            sets.append(trio)
```

Comment:
You should aim to write everything in a single line, even if you can. 
This link https://towardsdatascience.com/you-should-stop-craving-for-python-one-liners-5dbf8e0de23d explains it well, even though I disagree with how important memory and runtime are in most python programs.